libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -624,39 +624,36 @@ - (void)setOauthClientId:(NSString *)newClientId
 
 + (void)resetSessionCookie
 {
-    [self removeCookies:@[@"sid"]
-            fromDomains:@[@".salesforce.com", @".force.com", @".cloudforce.com"]];
-    [self addSidCookieForInstance];
+    __weak typeof(self) weakSelf = self;
```

Comment:
@khawkins @wmathurin We can do both for now i.e use the NSHttpCookieStorage and reset the pool. Ill make these changes.